Code/Tools/TestImpactFramework/Runtime/Common/Code/Source/Artifact/Factory/TestImpactTestEnumerationSuiteFactory.cpp

```diff
@@ -84,4 +87,74 @@ namespace TestImpact
+
+                // This regex matches pytest test names, in the form ScriptPath::TestFixture::TestName. It will accept both parameterized
+                // and unparamaterized tests (with or without [] at the end).
+                AZStd::basic_regex testNamePattern = AZStd::basic_regex("(.*py)::([A-Za-z_/\\-0-9]*)::(.*)");
+                AZStd::smatch matchResults;
+                if (AZStd::regex_search(curLine, matchResults, testNamePattern))
+                {
+                    AZStd::string absoluteScriptPath = matchResults[1];
+                    AZStd::string testFixture = matchResults[2];
+                    AZStd::string testName = matchResults[3];
+
+                    // Fetch or create the vector for this script
+                    AZStd::vector<TestEnumerationSuite>& suitesForThisScript = testSuiteMap[absoluteScriptPath];
+
+                    // If we're not working with the same fixture as our prevous iteration, or our suites vector is empty, create a test
+                    // suite, add the current test case to it, and add the suite to our vector.
+                    if (previousTestFixture != testFixture || suitesForThisScript.empty())
+                    {
+                        TestEnumerationSuite currentTestSuite =
+                            TestEnumerationSuite{ testFixture, true, AZStd::vector<TestEnumerationCase>() };
+                        currentTestSuite.m_tests.emplace_back(TestEnumerationCase{ testName, true });
+                        suitesForThisScript.push_back(currentTestSuite);
+                    }
+                    // Else, find the test suite in our vector, get the reference to it and add our current test case to it.
+                    else
+                    {
+                        auto matchTestSuite = [&testFixture](const TestEnumerationSuite& suite)
+                        {
+                            return suite.m_name == testFixture;
+                        };
+                        TestEnumerationSuite* currentTestSuitePointer =
+                            AZStd::find_if(suitesForThisScript.begin(), suitesForThisScript.end(), matchTestSuite);
+                        TestEnumerationSuite& currentTestSuite = *currentTestSuitePointer;
+                        currentTestSuite.m_tests.emplace_back(TestEnumerationCase{ testName, true });
+                    }
+
+                    // Update previoust test fixture so that we can keep track of when test fixtures change.
+                    previousTestFixture = testFixture;
+                    count++;
+                }
+            }
+
+            // Extract the key/value pairs from our testSuiteMap and put them in our pairs output variable.
+            for (const auto& [scriptPath, testSuiteVector] : testSuiteMap)
+            {
+                pairs.emplace_back(AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>(scriptPath, testSuiteVector));
```

Comment:
update: This noisy type (`AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>` appears in 4 places - consider moving this to a named alias - `using GoodTypeName = AZStd::pair<AZStd::string, AZStd::vector<TestEnumerationSuite>>` so if we ever change it it's simple to update in one place and can give more contextual information (we work at a higher level of abstraction)

Reply:
Can do! Where would you recommend this be placed within the file?

Reply:
Probably the .h file were the interface is first presented makes the most sense? Or if it is only used in the .cpp it could go in the top of that file (ideally it's good to limit the scope to only where it's needed if possible)

Reply:
Done!